### ISSUE TITLE
Allow for enabling privileged mode on containers

### ISSUE THREAD

**Issue by jersten** [docker/docker#24414]
Not a bug; more like a feature request.

I was running a container in non-privileged mode; I needed to change this container to privileged mode because it was running quagga. It would have been nice if I could have edited the already created containers config: `docker inspect --edit <container>`

There are other means too; maybe `docker update --privileged <container>` (boolean that just reverses the config that is in `docker inspect`).

**Comment by cpuguy83** [comment]
I don't think this would be possible while the container is live. Also, does quagga really need privileged, or just the NET_ADMIN cap?

**Comment by jersten** [comment]
Doesn't have to be live for me.

I was getting the following error:
```
ospfdprivs_init: initial cap_set_proc failed
```

Which led me [here](https://openvz.org/Quagga_in_VE). Then to enabling privileged mode on a container; after doing that quagga is able to start.

**Comment by cpuguy83** [comment]
With this error, I'd do `--cap-add SETPCAP` I believe.

Why not just re-create the container in this case?

**Comment by jersten** [comment]
One could just restart the container; personally, I'm seeing "privileged" mode as a resource that can/could/should be configurable.

**Comment by justincormack** [comment]
@jersten it can't really be, as you cannot apply capabilities to processes after they start, so you have to restart the container anyway, so I don't really think this is possible.

**Comment by thaJeztah** [comment]
Based on the above, I don't see us being able to implement this. Also; we decided to "curate" the list of option that can be changed through `docker update`, and only add options that make sense at runtime. If a container is started without `--privileged` or a capability that is required to run the container, that's basically an invalid configuration of the container, and not something that should be changed at runtime. Recreating the container with the right options is the right option for that.

I'm closing this issue, but feel free to comment after I closed
